File: src/com/sterndu/json/BoolValue.java
package com.sterndu.json;

import java.util.function.Function;

public class BoolValue implements JsonValue {

	private boolean value = false;

	public BoolValue(boolean value) {
		this.value = value;
	}

	public boolean getValue() { return value; }

	public void setValue(boolean value) { this.value = value; }

	@Override
	public String toJson() {
		return "" + value;
	}

	@Override
	public String toJson(Function<Object, String> function) {
		return "" + value;
	}

	@Override
	public String toString() {
		return toJson();
	}

}

File: src/com/sterndu/json/BoolValue.kt
@file:JvmName("BoolValue")
package com.sterndu.json

class BoolValue @JvmOverloads constructor(var value: Boolean = false) : JsonValue {

	override fun toJson(): String {
		return value.toString()
	}

	override fun toJson(function: (Any?) -> String): String {
		return value.toString()
	}

	override fun toJsonValue(): JsonValue {
		return this
	}

	override fun toJsonValue(function: (Any?) -> String): JsonValue {
		return this
	}

	override fun toString(): String {
		return toJson()
	}
}
